rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,108 @@
+---
+feature: NixOS Code of Conduct
+start-date: 2021-11-03
+author: @jonringer
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues: #98 #102
+---
+
+# Summary
+[summary]: #summary
+
+The NixOS Community needs some foundation of shared values which can be used
+to determine appropriate when contributing and interacting with the community.
+We currently have a [community ethos](https://nixos.org/community/index.html), but
+this only states that we are an inclusive community, but doesn't define
+inappropriate behavior when interacting with the community. This RFC aims
+to establish the shared values, so they can be used to identify behavior
+which is disruptive to the community.
+
+This RFC does not intend to define moderation practices. This is intended to only lay
+a framework of shared values for the community.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community struggles with a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only the moderation body name has been changed.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
+sexual orientation, disability, personal appearance, body size, race, ethnicity, age,
+religion, nationality, or other similar characteristics.
+- Please avoid using overtly sexual aliases or other nicknames that might
+detract from a friendly, safe and welcoming environment for all.
+- Please be kind and courteous. There’s no need to be mean or rude.
+- Respect that people have differences of opinion and that every design or
+implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
+- Please keep unstructured critique to a minimum. If you have solid ideas
+you want to experiment with, make a fork and see how it works.
+- We will exclude you from interaction if you insult, demean or harass anyone.
+That is not welcome behavior. We interpret the term “harassment” as including the definition in the
+[Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md);
+if you have any lack of clarity about what might be included in that concept,
+please read their definition. In particular, we don’t tolerate behavior that excludes
+people in socially marginalized groups.
```

Comment:
Separately, if we do «in particular» here, I would say that currently the most relevant thing moderated as harrassment is calling a person in a way they clearly asked not to be called, possibly save for polite references to the reasonably recent and currently relevant contributions/interactions.

Reply:
I'm not sure what you're asking from this RFC.

In the case of sandro asking not to be called SS2K (because SS), I think that's valid. But this would fall under harassment if that's the case.

Reply:
I want to make things that are implied-bad and have lead to multiple bans to be as prominently discouraged as reasonable. And as far as I remember multiple people have been banned for deliberately and explicitly insisting on using some pronouns in reference to specific people in the community after being asked not to use those.

So I would suggest adding «Neither do we tolerate calling people how they asked not to be called.» and then I guess someone would suggest a better wording.

I agree that Sandro's request was also perfectly reasonable, of course. Fortunately, there things went more amicably.

Reply:
I think that should be left up to the moderation to arbitrate those incidents.

I think we are getting CoC and enforcement of CoC blurred here. If someone is specifically using the wrong pronoun to harass someone, then that's harassment; and the moderation team should be able to enforce it.

Reply:
«to harass» is an intent, and the logic of those who insist on some kind of essentialism implying the use of the rejected pronouns is alien enough that reaching understanding of intent in any direction is hard… I have seen people go overboard in the other direction, too (explicitly claiming that deliberate avoidance of _all_ pronoun use in some space is harassment via non-recognition of pronoun preference). 

That's why I think calling out a modal example of what actually is not to do (however violations are handled, that's moderation discretion up to a point) could put the intent into a more inter-subjective light.

Reply:
> «to harass» is an intent, and the logic of those who insist on some kind of essentialism implying the use of the rejected pronouns is alien enough to those who request specific pronouns that reaching understanding of intent in any direction is hard…

genesis (IRC) was banned because he was a non-native speaker and struggled with the use of "they" to refer to a single individual. He was using the [gender neutrl he/him](https://en.wikipedia.org/wiki/Gender_neutrality_in_languages_with_gendered_third-person_pronouns#Generic_he_and_she) instead of the preferred they/them. [The cognitive dissonance](https://logs.nix.samueldr.com/nixos/2020-03-14#3177147), and [subsequent "temp" ban](https://logs.nix.samueldr.com/nixos-dev/2020-03-14#3177453). Difficulty in understanding a world view and language diction which differs from the one you are familiar should not be grounds for removal. For more context, in french "vous" is sometimes used to refer to God. "vous" is also used to refer to [one or more people](https://ai.glossika.com/blog/when-to-use-tu-and-when-to-use-vous) and is most similar to the english "they". An individual wanting to be called "they" could be seen as wanting to be called "God". (I'm intimate with the french language, please correct me if I'm wrong. But it seems to be that genesis struggled with the usage of "they", but didn't mean to cause insult to another contributor).

This is just one example where making pronouns a priority can exclude people that aren't intimate with 21st century American-English and its nuances.

>  I have seen people go overboard in the other direction, too (explicitly claiming that deliberate avoidance of all pronoun use in some space is harassment via non-recognition of pronoun preference).

I'm guilty of this as well. I interact with a lot of strangers on the internet. Unless their pronouns are part of their username, I'm not going to remember. So avoiding pronouns is preferred over incorrect usage.

In an online setting, especially Nixpkgs, I don't think it's reasonable to enforce strangers to be aware of other stranger's pronouns (unless explicit e.g. after the username). If the usage is meant to cause harm, then that falls under the other statutes.

> That's why I think calling out a modal example of what actually is not to do (however violations are handled, that's moderation discretion up to a point) could put the intent into a more inter-subjective light.

Yes, but it's a very specific use case. And the actual disruptive behavior is covered by other statues.

Reply:
Yeah, I also reduce the amount of personal pronouns I use.

About genesis I would say they made some further comments on the topic (on IRC), with the tone that might have affected Graham's decisions. And sure there are pronoun choices that many will not want to follow even if the choice is there in the display name (or even couldn't? what if diacritics is relevant?), which is fine. 

I think the current wording of the guidelines does not _exclude_ the interpretation that writing «calls himself ‘they’» is already outright bannable harassment (this is stronger than the position actually applied in the moderation decision in question), and it also does not clearly _imply_ that weird requests about pronouns should be treated with complete avoidance of pronouns in favour of nick autocomplete (as long as you honestly remember). 

As this is literally an issue where GitHub moderators and Discourse moderators have disagreed in the past, I find it useful to narrow the range of interpretations at least a bit.

Reply:
I think we should remember that the entire pronouns topic is still only really common in the left bubble. Someone not familiar with that will most likely be really confused and do some mistakes even if they don't intent to hurt anyone. I think that is really important to remember. Also NixOS and nixpkgs are first a distro and a package collection, not a social movement and we should choose our fights and not fight every front with radical ideas.

Reply:
@SuperSandro2000: What is the radical idea that you're referring to?

Reply:
Honestly, as a relatively new NixOS/nixpkgs contributor, this is really disheartening to see.

Personally speaking, seeing a prominent contributor express the opinion that using someone's pronouns is a "radical" idea that is "only really common in the left bubble" instead of _basic decency_ is incredibly demoralizing. And the fact that (judging by the reactions) both of the current release managers seem to agree with this sentiment almost makes me want to avoid the project entirely.

I strongly believe that there is no apolitical software. NixOS and nixpkgs are more than a distro and a package collection—they are part of a _community_ of people who contribute to and use them. And this community is made up of real people with real lives, not abstract entities.

Unfortunately, there are no neutral positions here; one either tolerates intolerance in their community or they don't. And your comment seems to imply the former—that, when forced to choose, you would rather have people who are intolerant of others in the community than those they will inevitably push out.

I know my opinion here carries relatively little weight, I just wanted to share my own perspective. I know I can't force you to reconsider, however I would ask that if this choice is made that it be clearly stated in the code of conduct so that new members of the community can be aware of it.

Reply:
unfamiliar/confused != intolerance

Reply:
The comment I am responding to seems to be a statement on "the entire pronouns topic". It is my understanding that this includes people using the wrong pronouns in _all_ situations, including intentionally. It is possible that I misunderstood, however even then I would object to the argument that NixOS/nixpkgs is not a "social movement" being used to justify community moderation decisions, which are inherently social.

Reply:
Moderation of social activities could have a goal of social harmony without having a goal of social change. The more modest goal of social harmony might still require social change on the part of some community members, but it would be a lesser degree.

Reply:
If I understand that correctly then I think I agree. I guess the question then is is who is responsible for changing? Personally, in a situation where someone is being referred to in a way they dislike, I can see two possibilities: either they can decide they are willing to put up with being referred to that way, or the other person can change how they are referring to them. I strongly believe that the latter is the best course of action.

More generally, I think I should probably clarify my stance on this. I would certainly hope that we are not punishing people for honest mistakes, especially since some things (like singular they) can be confusing to non-native speakers. Nor do I expect everyone to use singular they for people they don't know, as nice as that would be. My main concern is people who continue to _intentionally_ refer to someone a certain way after being asked not to. That, from my perspective, is clearly harassment.

The comment I initially replied to seemed to be taking a stance on using others' pronouns as a whole, which is why I responded strongly. Again, it is possible I misunderstood. I certainly hope I did, as that would mean that there is no disagreement here, but I don't know if that is the case right now.

Reply:
@InternetUnexplorer I interpret the original comment as highlighting that NixOS is global, linguistically and culturally diverse, and that we should be inclusive to the best extent possible. I believe we agree about the overall end-goal, to not allow harassing behavior. Perhaps changing the topic and setting slightly would provide a way to analyze a hypothetical. It is a common bullying tactic to use unkind nicknames or diminutives as a way to cause repeated verbal harm. At some point, repeated incorrect usage of someone's name in a particular venue - in spite of being corrected - rises to the level of harassment. At the same time, it would also be harassment to deliberately follow (see discussion above about intimidation, stalking, following) someone to every online venue to correct their language, depriving them of the equivalent of personal space. The overall goal is to reduce harassment; hence the reference to another document which attempts to provide additional commentary.

The point of contention, and it seems this is the more fundamental question, is about neutral positions. Are there neutral positions we take? If the focus is to remain on the distro, package collection, and related technologies, how do we handle the case where some view a position as a binary with no middle or neutral ground?

Reply:
I agree that repeatedly using the wrong name for someone would be similar to using the wrong pronouns, even if the two situations are not quite the same. And I think that the point at which it becomes harassment would be very early on; once someone has been explicitly asked not to refer to someone a certain way, to continue to do so seems to me like a deliberate choice. Of course, mistakes happen now and then, but in my personal experience the difference between making an effort and not caring at all is large enough that it's pretty easy to tell which is which.

As for neutral positions, I really do believe that a decision like this is binary. Either one stands up against this behavior when they see it happening or they don't. To stay out of the situation is a decision to allow that behavior; there isn't really a way they can take a "neutral" position of both allowing it and not allowing it. Of course, where exactly the line is that determines whether something is harassment is much more difficult to decide; I didn't mean to sound like I was implying that decision was binary.

The comment I initially responded to seemed, to me, to be taking the position that there was a neutral option in the matter, and that the neutral option was to allow it/the non-neutral or "radical" position was to not allow it.

Specifically, it said that "the entire pronouns topic is only really common in the left bubble", which seems to me to be saying that a) the "center" or "neutral" position would be to not discuss the entire pronouns topic at all, and b) that those who do are in a bubble, i.e. that their understanding of reality is warped or not indicative of the majority. That, specifically, is what I was referring to when I said I found it disheartening.

Reply:
I think there is value in explicitly stating that we consider it unacceptable to repeatedly refer to someone in a way they have asked not to be referred to.

> We will exclude you from interaction if you insult, demean or harass anyone.

I think this covers both cases of insulting or demeaning nicknames as well as repeatedly using the wrong pronouns to refer to someone.

> Sexist, racist, homophobic, transphobic, ableist or otherwise discriminatory jokes and language.

From the linked Citizen Code of Conduct, I think this bullet would also cover misgendering someone by using the wrong pronouns repeatedly.

In both of the above cases, however, I don't know if my interpretation is universal. The question of whether we need to use people's pronouns seems to keep coming up, so my opinion is that we should have a clear stance on the subject with our reasoning rather than continuously re-litigate it. I hope it would also make it easier to explain to people with less experience in English why we try to respect people's preferred pronouns even if it may not be what they are used to.

> So I would suggest adding «Neither do we tolerate calling people how they asked not to be called.» and then I guess someone would suggest a better wording.

I would suggest "It is unacceptable to repeatedly refer to someone in a way that they have asked not to be referred to. We strive to use people's chosen names and pronouns to show them that they are respected and welcome."

Hopefully this communicates that mistakes happen, and we understand using the wrong pronoun or nickname by accident is not serious. It may be better to replace "repeatedly" with "intentionally", but I don't want to litigate anyone's intentions and it would be the only place in the document to focus on intention (which we can't see) instead of action. I hope this doesn't make it seem like we expect everyone to magically know everyone else's pronouns at all times, the second sentence is meant to be in the same spirit as other parts of the document that provide reasoning such as "There is seldom a single right answer" or that a rule exists to ensure a "welcoming environment for all."

Reply:
> I would suggest "It is unacceptable to repeatedly refer to someone in a way that they have asked not to be referred to. We strive to use people's chosen names and pronouns to show them that they are respected and welcome."

I would slightly weaken the latter part.  «We strive to follow people's preference on names and pronouns as a part of basic expectations of politeness.» On the one hand, even if you as a moderator ban a person, you cannot convincingly claim you still welcome the troll at that specific moment, but should not intentionally misgender the banned person in the process. On the other hand, everyone chooses which option consistent with the preferences to pick. I will not make an effort to look up and use pronouns when I am in conversation with fellow contributors I barely know, I will probably just refer to everyone (including those in the conversation I do know, or those who somehow display the pronouns) by the usernames they themselves have chosen on the platform (or by displaynames, depending what the platform shows more prominently).

Reply:
> genesis (IRC) was banned because he was a non-native speaker and struggled with the use of "they" to refer to a single individual. He was using the [gender neutrl he/him](https://en.wikipedia.org/wiki/Gender_neutrality_in_languages_with_gendered_third-person_pronouns#Generic_he_and_she) instead of the preferred they/them. [The cognitive dissonance](https://logs.nix.samueldr.com/nixos/2020-03-14#3177147), and [subsequent "temp" ban](https://logs.nix.samueldr.com/nixos-dev/2020-03-14#3177453).

I don't think this is true.

The message directly after the one you link is genesis saying "my self i'm a blue wolf with 2 pink nose and one green big finger."  My interpretation of this is that it's a transphobic comment attempting to ridicule WORLDofPEACE's non-binary identity.  It doesn't make sense that somebody who's just confused about how to speak English but otherwise engaging in good faith would come out with a comment like this, because it's not about English, it's about identity, specifically trans identity.

Of course it doesn't make sense to ban somebody for making language mistakes, but that's not what happened here.  I can think of at least one case where a non-native English speaker was told that "he" was not an appropriate way to refer to WORLDofPEACE, and hadn't come across the concept of singular they before.  In that case, they made it clear they didn't mean any disrespect, and there was never a problem with that person again.

Reply:
This is closed and shut already:

> 21:28 <genesis> yes adisbladis but i'm not the one to convinct here
> 21:28 <genesis> but the "THEIR"

Calling WoP "the 'THEIR'" this way is proof of malice you directly linked to. Especially if you know french and know the badly translated turn of phrase used here. "the 'THEIR'" here was clearly used maliciously to refer not to WoP the person, but to WoP "the pronoun using [person]". This, to me, clearly shows they understood.

Note: "convinct" here could be mistaken for "convict". In this exchange it's "convince" (convaincre). In reference to WoP's  reviews on the PR.

Reply:
(Now reading the full comment partially quoted, for due diligence.)

While I totally agree, and expect any "enforcement" to consider the background of the individual, this case wasn't one of those. It would be hypocritical to not consider the background of the individual when considering issues, as *things are different in different locales*.

Now, about the content of the discussion:

> For more context, in french "vous" is sometimes used to refer to God.

I don't think this is right. Though maybe it's a way older deprecated usage that's been continuously taught in french as a second language, or something that didn't stick when the language travelled to the Americas. With that said I couldn't find confirmation about "vous" being used to refer to god anywhere in my usual language resources.

Though, what I had in mind was that "il" (he) sometimes is used to refer to god, but as a narrative construct. Not as a form of respect. But I couldn't confirm either as I wasn't raised in that kind of background. (Or maybe it's "ils", but that wouldn't make sense in a monotheistic religion.)

> "vous" is also used to refer to one or more people and is most similar to the english "they".

Not exactly. It's one person *in polite context*. It's "tu" (you), but polite. It's not even the most similar to the english "they", it's most similar to "you", where in english "you" is both singular and plural all the time.

> An individual wanting to be called "they" could be seen as wanting to be called "God". (I'm not intimate with the french language, please correct me if I'm wrong. But it seems to be that genesis struggled with the linguistic usage of "they" to mean an individual, so the individual must therefore be more than just a single individual (e.g. collective), but didn't mean to intend to cause insult to another contributor).

As I said, I can't disprove that it would be a reference to god in some context, but I wasn't able to find confirmation in my usual language resources.

The more correct (and charitable) interpretation is *they* will be interpreted as "ils/elles", which is always plural, and gendered still. This can make internalizing the differences harder.

This particular case is harder to discern as it's both a gender neutral pronoun, and a plural pronoun use. Most discussions around "they", and gender neutrality has been around "singular people".

So anything about a language you don't master should be taken with a pound of salt. As I do in every languages.

As a final note, while yes initially the confusion has to be tolerated from native french speaking people, **continuous** use of the wrong pronoun cannot be, and then **ridiculing** cannot be tolerated.

Reply:
> Personally speaking, seeing a prominent contributor express the opinion that using someone's pronouns is a "radical" idea that is "only really common in the left bubble" instead of basic decency is incredibly demoralizing.

That's not what I wanted to convey. I wanted to keep my comment short which afterwards left it open for to much interpretation. Sorry for that. Singular they and to more extend neo pronouns is something which I personally don't feel comfortable with expecting from everyone to be fluent in. To make it clear: It is not acceptable to use someones "born with" pronoun that is otherwise completely not fitting.

> As for neutral positions, I really do believe that a decision like this is binary. Either one stands up against this behavior when they see it happening or they don't.

If you are not familiar enough with a topic, don't have enough information or didn't think much about it yet there is a third choice: I don't have an opinion about that, I am not decided yet. We need to remember that not everyone is highly informed about the social issues of our time which is not great but okay.

> especially since some things (like singular they) can be confusing to non-native speakers

That is my point. We have a really high variance of English skills in our community and I don't think we consider this enough. I would rate my English understanding quite high but my reproduction skills and especially my grammar is not that high and more average. My German grammar is also not that great but for example: I can write a gender neutral texts in a good speed but I can't do that in English. My skills are not good enough to write a longer text in a gender neutral way and correct grammar and we are putting really high pressure on people which do not have such a high  skill level.

If you intentionally miss gender someone with he/she/etc in your native language you are quite accountable for that but I don't believe we should put that on a level with people for which English is the second/third/etc language and they are still learning it.

> My main concern is people who continue to intentionally refer to someone a certain way after being asked not to. That, from my perspective, is clearly harassment.

That is not what I try to discuss. Of course that is not okay, we are no longer in a school yard but not every occurrence of this is immediately harassment.

> Specifically, it said that "the entire pronouns topic is only really common in the left bubble", which seems to me to be saying that a) the "center" or "neutral" position would be to not discuss the entire pronouns topic at all and b) that those who do are in a bubble, i.e. that their understanding of reality is warped or not indicative of the majority. That, specifically, is what I was referring to when I said I found it disheartening.

That was not phrased correctly from me. There are basically three stances on this:
- The (far) right which talk the entire day about pronouns and gendering destroying the nature of the language and it will be end of the country and civilization as we know it.
- The neutral ground which don't discuss it at all.
- The left which accepted pronouns and gendering and don't talk to much about it.

> I hope this doesn't make it seem like we expect everyone to magically know everyone else's pronouns at all times

I think this can be interpreted like this which is a problem. Also checking someones profile description before answering a question feels like stalking.

Siamo tutti antifascisti ✊🏼 🏴

Reply:
It seems I misunderstood your comment, then. Apologies and thank you for clarifying!

Reply:
Not a problem. That's why we are talking about and that's why I prefer conversation in real time and person because there you can briefly interrupt someone if there is a clear misunderstanding of what you intended to say.